python/ray/data/_internal/planner/plan_expression/expression_visitors.py

```diff
@@ -347,3 +387,92 @@ def visit_download(self, expr: "DownloadExpr") -> str:
 
     def visit_star(self, expr: "StarExpr") -> str:
         return self._make_tree_lines("COL(*)", expr=expr)
+
+
+class _IcebergExpressionVisitor(_ExprVisitor["BooleanExpression"]):
```

Comment:
I found it slightly unexpected that we're placing Iceberg-specific code in the general `expression_visitors.py` module. Might be more maintainable if we colocate it with the Iceberg code